app/controllers/customers_controller.rb

```diff
@@ -1,76 +1,79 @@
-class CustomersController < ApplicationController
-  rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
-  before_action :set_customer, only: %i[show edit update destroy]
-
-  # GET /customers or /customers.json
-  def index
-    @customers = Customer.all
-  end
-
-  # GET /customers/1 or /customers/1.json
-  def show
-  end
-
-  # GET /customers/new
-  def new
-    @customer = Customer.new
-  end
-
-  # GET /customers/1/edit
-  def edit
-  end
-
-  # POST /customers or /customers.json
-  def create
-    @customer = Customer.new(customer_params)
-
-    respond_to do |format|
-      if @customer.save
-        format.html { redirect_to customer_url(@customer), notice: "Customer was successfully created." }
-        format.json { render :show, status: :created, location: @customer }
-      else
-        format.html { render :new, status: :unprocessable_entity }
-        format.json { render json: @customer.errors, status: :unprocessable_entity }
-      end
-    end
-  end
-
-  # PATCH/PUT /customers/1 or /customers/1.json
-  def update
-    respond_to do |format|
-      if @customer.update(customer_params)
-        format.html { redirect_to customer_url(@customer), notice: "Customer was successfully updated." }
-        format.json { render :show, status: :ok, location: @customer }
-      else
-        format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @customer.errors, status: :unprocessable_entity }
-      end
-    end
-  end
-
-  # DELETE /customers/1 or /customers/1.json
-  def destroy
-    @customer.destroy
-    respond_to do |format|
-      format.html { redirect_to customers_url, notice: "Customer was successfully destroyed." }
-      format.json { head :no_content }
-    end
-  end
-
-  private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_customer
-      @customer = Customer.find(params[:id])
-    end
-
-    # Only allow a list of trusted parameters through.
-    def customer_params
-      params.require(:customer).permit(:first_name, :last_name, :phone, :email)
-    end
-
-    def catch_not_found(e)
-      Rails.logger.debug("We had a not found exception.")
-      flash.alert = e.to_s
-      redirect_to customers_path
-    end
-end
+class CustomersController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
+  before_action :set_customer, only: %i[show edit update destroy]
+
+  # GET /customers or /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/1 or /customers/1.json
+  def show
+    @customer = Customer.find(params[:id])

```

Comment:
`set_customer` should already be setting the instance variable, so I don't think you need this line.